### PR TITLE
Credentials into docker compose

### PR DESCRIPTION
**Pull request by samuelbernardolip**
Pass credentials into docker compose using arguments provided to exec comand.
resolves PR #65 PR #82

### CODE REVIEW

File: src/eu/indigo/compose/DockerCompose.groovy [hunk 1]
```diff
@@ -121,14 +138,21 @@ class DockerCompose extends JenkinsDefinitions implements Serializable {
...
     */
-    List<String> envToStep(Map env) {
-        env.collect { e, v ->
-            "$e=\"$v\""
+    String getCredsVars() {
+        String res = ''
+
+        if (_DEBUG_) { steps.echo "credentialVariablesNames:\n${credentialVariablesNames}" }
+
+        if (! credentialVariablesNames?.empty) {
```
Comment (orviz) [inline src/eu/indigo/compose/DockerCompose.groovy:149]: the `collection?.empty` statement seems not to work, use `isEmpty()` instead:

```
-        if (! credentialVariablesNames?.empty) {
+        if (! credentialVariablesNames.isEmpty()) {
             credentialVariablesNames.each { v ->
                 res += "-e ${v} "
             }
```

File: src/eu/indigo/compose/DockerCompose.groovy [hunk 2]
```diff
@@ -121,14 +138,21 @@ class DockerCompose extends JenkinsDefinitions implements Serializable {
...
     *
-    * @param env Map with the variable name as key and the expected value
     */
-    List<String> envToStep(Map env) {
-        env.collect { e, v ->
-            "$e=\"$v\""
+    String getCredsVars() {
+        String res = ''
+
+        if (_DEBUG_) { steps.echo "credentialVariablesNames:\n${credentialVariablesNames}" }
```
Comment (orviz) [inline src/eu/indigo/compose/DockerCompose.groovy:147]: the debug msg should include at least the method name, so we know where this msg comes from. Is it worth using a log library like `Slf4j`? or do we just keep it simple like this?

Comment (samuelbernardolip) [reply src/eu/indigo/compose/DockerCompose.groovy:147]: Jenkins framework have already is own logging system.
Using an external logging library is very probably limited by the Jenkins security sandbox limitations.
I already placed a simple implementation in JenkinsDefinitions class for loglevel and also keeping the development debug level.

**Comment by samuelbernardolip** [comment]
@orviz There is a bug since credentials variable appears duplicated.
We need to solve that before doing the merge.
This PR already as the changes from test/credentials-into-docker-compose, so this branch can also be deleted in the end.

**Comment by orviz** [comment]
just removed `test/credentials-into-docker-compose` branch

**Comment by orviz** [comment]
There was an extra call to `credentialsToStep()` method in a DEBUG line. The last commit just comments this line out.

I've tried to store the return value of `credentialsToStep()` so to be used both for the debugging and for the `withCredentials` statement, but when I do that I get a complain from Jenkins script security (`org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method groovy.lang.GroovyObject setProperty java.lang.String java.lang.Object`)

**Pull request merged**